src/client/dfs/duns.c

```diff
@@ -582,6 +592,49 @@ duns_create_path(daos_handle_t poh, const char *path, struct duns_attr_t *attrp)
 			rc = daos_cont_create(poh, attrp->da_cuuid, prop, NULL);
 			daos_prop_free(prop);
 		}
+
+		if (rc == -DER_SUCCESS && backend_dfuse) {
+			/* Setup the DFUSE entry points.  Do this after the
+			 * container has been created as the first access after
+			 * setting these will do a container attach and that
+			 * will fail if the container doesn't exist.
+			 *
+			 * Set both the extended attribures the DFUSE expects
+			 * as well as the Lustre version, which is used by
+			 * duns_resove_path() in container destroy.
+			 */
+
+			rc = lsetxattr(path, "user.uns.pool", pool,
```

Comment:
so the str in the  DUNS_XATTR_NAME, already container the pool and container uuid. why are we doing each of the pool and cont uuids in a separate xattr, then storing them again also in the DUNS format?
what if the original lsetxattr executed before also succeeded? then we have 2 different sets of xattrs.. i guess im also confused.

Reply:
dfuse and Lustre/duns use different xattrs.  The dfuse ones are set for dfuse to work, the duns one is also set for the duns code in "container delete" to work.  I tried to explain this in the comment above.

Ideally we would remove the dfuse xattrs and just use the same set everywhere but simply using both was a quicker solution to get the feature landed.

Reply:
sorry i should've read the comments.
but why not change dfuse in this case to read and parse the extended attribute that is set on line 566 rather than needing 2 extra extended attributes here. i guess this might be what you are saying, i was just wondering why in the first place you didn't go that route.